syntax = "proto3";

package ai;

option go_package = "climbinsight/server/ai;ai";

// サービス定義：AIService という名前のサービスを定義する
service AIService {
  // Process メソッドを定義する。リクエストを受けてレスポンスを返す。
  rpc Process(InputRequest) returns (ProcessImageResponse);
}

// クライアントから送られてくるリクエストデータ
message InputRequest {
  bytes image = 1; // フィールド名: input（1番目）
  repeated Point points = 2;
}

message Point {
  float x = 1;
  float y = 2;
}

// サーバーが返すレスポンスデータ
message ProcessImageResponse {
  bytes processed_image = 1; // RGBA PNG画像
  string mime_type = 2;      // "image/png"
}
